### PR TITLE
feat: Implemented gRPC server for authorization 

### PR DESCRIPTION
**Pull request by ashish493**
@hsluoyz , do we need logo or docs for this repo?

### CODE REVIEW

File: authz.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		glog.Errorf("Filed to load the policies: %v", err)
+		return
+	}
+
+	user, _, _ := req.BasicAuth()
+	method := req.Method
+	path := req.URL.Path
+
+	if e.Enforce(user, path, method) {
```
Comment (ashish493) [inline authz.go]: This is the authorization logic.

**Comment by hsluoyz** [comment]
@ComradeProgrammer plz review

**Comment by hsluoyz** [comment]
> @hsluoyz , do we need logo or docs for this repo?

of course

File: authz.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+								Key:   "casbin-authorized",
+								Value: "allowed",
+							},
+						},
+					},
+				},
+			},
+		}, nil
+	}
+	else {
```
Comment (ComradeProgrammer) [inline authz.go]: Can this code compile? I remember that putting 'else' in a new line violates go grammar.

File: authz.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+				DeniedResponse: &auth.DeniedHttpResponse{
+					Status: &envoy_type.HttpStatus{
+						Code: envoy_type.StatusCode_Unauthorized,
+					},
+					Body: "PERMISSION_DENIED",
+				},
+			},
+		}, nil
+
+	}
```
Comment (ComradeProgrammer) [inline authz.go]: Does the bracket match?

File: authz.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+type AuthorizationServer struct{}
+
+func (a *AuthorizationServer) Check(ctx context.Context, req *auth.CheckRequest) (*auth.CheckResponse, error) {
+	log.Println(">>> Server Performing authorization check!")
+	
+	e, err := casbin.NewEnforcer("./example/model.conf", "./example/policy.csv")
+	if err != nil {
+		glog.Errorf("Filed to load the policies: %v", err)
+		return
```
Comment (ComradeProgrammer) [inline authz.go]: Is it legal to return 0 values while the declaration demands 2 return values?

File: authz.go (hunk 2 above)
Comment (ashish493) [reply authz.go]: Haven't checked it, I will check with go lint for this.

File: authz.go (hunk 3 above)
Comment (ashish493) [reply authz.go]: Let me use gofmt to align the brackets.

File: authz.go (hunk 4 above)
Comment (ashish493) [reply authz.go]: My bad. I actually wanted to use `log.Fatalf("Failed to load the policies:: %v", err)`. I will replace it and remove the return.

**Comment by ashish493** [comment]
@ComradeProgrammer , took some time, but now it is compiling and working fine.

**Review by ComradeProgrammer — APPROVED** [review]

**Pull request merged**